book/modules/mod-A/nav.adoc

```diff
@@ -1,3 +1,5 @@
+:module-name: {page-module}
```

Comment:
This page attribute will be passed to the  `modules/shared/partials/common-nav.adoc` file, making import its shared pages dynamically from this module. Note that page attributes inside `include` -ed pages are evaluated within the context of the page that does the including. 

This means that when inside `mod-A/nav.adoc` we call:

```asciidoc
include::shared:partial$common-nav.adoc[]
```

And inside `modules/shared/partials/common-nav.adoc` we have:

```asciidoc
* xref:{module-name}:shared/common-page.adoc[Link to common page]
```

The value of `{module-name}` when the `xref` is evaluated will be the value of `{page-module}` inside `modules/mod-A/nav.adoc`, which is `mod-A` (and not its value in the `shared` module, which is `shared`). This in combination with the symlinks makes the xref "dynamically" point to the module that is importing it at site build time.